lib/v2/sspu/pe.js

```diff
@@ -0,0 +1,63 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const { id = '342' } = ctx.params;
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit, 10) : 30;
+
+    const rootUrl = 'https://pe2016.sspu.edu.cn';
+    const currentUrl = new URL(`${id}/list.htm`, rootUrl).href;
+
+    const { data: response } = await got(currentUrl);
+
+    const $ = cheerio.load(response);
+
+    let items = $('table.wp_article_list_table a[title]')
+        .slice(0, limit)
+        .toArray()
+        .map((item) => {
+            item = $(item);
+
+            return {
+                title: item.text(),
+                link: new URL(item.prop('href'), rootUrl).href,
+                pubDate: parseDate(item.prev().text()),
+            };
+        });
+
+    items = await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const { data: detailResponse } = await got(item.link);
```

Comment:
Please block requests to binary files
https://github.com/DIYgod/RSSHub/actions/runs/7365081002/job/20046055879#step:12:19

Reply:
👌